integ/scripts/bash/run-e2e-automated.sh

```diff
@@ -30,19 +32,30 @@ refreshcreds() {
     unset CREDS
 }
 
+ASSUME_ROLE="false"
+if [ ! -z ${ROLE_ARN+x} ] && \
+   [ ! -z ${ROLE_SESSION_NAME+x} ] && \
+   [ ! -z ${ROLE_EXTERNAL_ID+x} ]; then
+  ASSUME_ROLE="true"
+fi
+
+if [ $ASSUME_ROLE = "true" ]; then
+  # Setup the hook that runs before any interactions with AWS to refresh the credentials being used.
+  # There is a 1 hour timeout on these credentials that cannot be adjusted.
+  export -f refreshcreds
+  export PRE_AWS_INTERACTION_HOOK=refreshcreds
+
+  refreshcreds
+fi
+
 # Basic integ test configuration
 export SKIP_TEST_CONFIG=true
 export DEADLINE_VERSION="10.1.11.5"
 
-# Setup the hook that runs before any interactions with AWS to refresh the credentials being used.
-# There is a 1 hour timeout on these credentials that cannot be adjusted.
-export -f refreshcreds
-export PRE_AWS_INTERACTION_HOOK=refreshcreds
-
-refreshcreds
-
 ./scripts/bash/rfdk-integ-e2e.sh
 
-unset AWS_ACCESS_KEY_ID
-unset AWS_SECRET_ACCESS_KEY
-unset AWS_SESSION_TOKEN
\ No newline at end of file
+if [ $ASSUME_ROLE = "true" ]; then
+  unset AWS_ACCESS_KEY_ID
+  unset AWS_SECRET_ACCESS_KEY
+  unset AWS_SESSION_TOKEN
+fi
```

Comment:
Missing new line, looks like this was a miss by me when I originally wrote it.

Reply:
Ah good catch. Added